lib/resty/websocket/client.lua

```diff
@@ -147,28 +155,41 @@ function _M.connect(self, uri, opts)
                 return nil, "custom headers must be a table"
             end
         end
+
+        host = opts.host
+        if host ~= nil and type(host) ~= "string" then
+            return nil, "custom host header must be a string"
+        end
     end
 
     local ok, err
     if sock_opts then
-        ok, err = sock:connect(host, port, sock_opts)
+        ok, err = sock:connect(addr, port, sock_opts)
     else
-        ok, err = sock:connect(host, port)
+        ok, err = sock:connect(addr, port)
     end
     if not ok then
         return nil, "failed to connect: " .. err
     end
 
-    if scheme == "wss" then
-        if not ssl_support then
-            return nil, "ngx_lua 0.9.11+ required for SSL sockets"
-        end
+    if ssl then
         if client_cert then
             ok, err = sock:setclientcert(client_cert, client_priv_key)
             if not ok then
                 return nil, "failed to set TLS client certificate: " .. err
             end
         end
+
+        if not server_name then
+            if host then
+                -- strip trailing port number
+                server_name = re_gsub(host, ":[0-9]+$", "", "oj")
```

Comment:
In pure OpenResty fashion we should try and avoid edge-cases, especially if using the ngx.re API for this feels way overkill (e.g. `string.find` + `string.sub` suffices and is much faster). So, in pure OpenResty fashion: if the user provided `server_name` or `host`, then the SNI is that: it will match the Host header which is ultimately what most servers want in their implementation. Otherwise, default to `addr` indeed. So in short this would suffice for example:
```lua
local server_name = opts.server_name or host or addr
```

Or to save the `type` call:
```lua
if not server_name then
    server_name = host or addr
end
```

Reply:
Hmm, I must have misunderstood your other comment regarding SNI + port number. I thought you were saying we should remove the port number from `server_name` if sourcing it from `opts.host` (therefore only allowing it to potentially contain a port number if the user explicitly passes in `opts.server_name = "hostname:1234"`). What I had before was more or less the same as your example here:

```lua
local server_name

[[ ... ]]

if opts then
    [[ ... ]]
    server_name = opts.server_name
    if server_name ~= nil and type(server_name) ~= "string" then
        return nil, "SSL server_name must be a string"
    end
end

if ssl then
  server_name = server_name or host or addr
  [[ ... ]]
end
```

Reply:
Oops, that's my bad for misunderstanding where the value was coming from the test. Glancing at it I thought it was coming from the concatenation or addr + port, I did not realize that stripping the port would involve parsing the host value but since it does, I think it isn't worth it - at least that's how I'd see OpenResty accepting this patch (in your previous form).

Reply:
:+1: all good; reverted back to no parsing/port-handling